site/src/pages/WorkspacesPage/batchActions.ts

```diff
@@ -62,9 +78,14 @@ export function useBatchActions(options: UseBatchActionsProps) {
 		},
 	});
 
+	// Not a great idea to return the promises from the Promise.all calls below
+	// because that then gives you a void array, which doesn't make sense with
+	// TypeScript's type system. Best to await them, and then have the wrapper
+	// mutation function return its own void promise
+
 	const favoriteAllMutation = useMutation({
-		mutationFn: (workspaces: readonly Workspace[]) => {
-			return Promise.all(
+		mutationFn: async (workspaces: readonly Workspace[]) => {
```

Comment:
I think I'd rather just specify a `: Promise<void>` return type to prevent it from getting changed back, and get rid of the comment. it's hopefully clear enough with just that why returning `Promise<Array<void>>` is weird.